init/setup-rwdev.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+set -e
+
+if [ -e /dev/xvdb ] ; then
+    # The private /dev/xvdb device is present.
+
+    # check if private.img (xvdb) is empty - all zeros
+    private_size_512=`blockdev --getsz /dev/xvdb`
+    if dd if=/dev/zero bs=512 count=$private_size_512 2>/dev/null | diff /dev/xvdb - >/dev/null; then
+        # the device is empty, create filesystem
+        echo "Virgin boot of the VM: creating private.img filesystem" >&2
+        mkfs.ext4 -m 0 -q /dev/xvdb || exit 1
+    fi
+
+    tune2fs -m 0 /dev/xvdb
+    echo "Virgin boot of the VM: marking private.img as clean" >&2
+    fsck.ext4 -fp /dev/xvdb
```

Comment:
Okay, I got what your question was.  Here's the deal: Qubes OS already did the resize management every time the system booted.  This was not a "oh, the device was created now, resize it".  Nono, the resize management needs to happen on boot because people may enlarge the device prior to the boot of the system, and this is the mechanism used to provide for that OS-perspective resize.

So why fsck.ext4?

No more.

I have enhanced it to do the fsck only when the resize fails ,because the resize operation demands the file system be marked as clean.

There you go.